cartographer_ros/cartographer_ros/map_builder_bridge.cc

```diff
@@ -257,6 +258,62 @@ MapBuilderBridge::GetLocalTrajectoryData() {
   return local_trajectory_data;
 }
 
+void MapBuilderBridge::HandleTrajectoryQuery(
+    cartographer_ros_msgs::TrajectoryQuery::Request& request,
+    cartographer_ros_msgs::TrajectoryQuery::Response& response) {
+  const auto trajectory_states = GetTrajectoryStates();
+  if (!trajectory_states.count(request.trajectory_id)) {
+    response.status.code = cartographer_ros_msgs::StatusCode::INVALID_ARGUMENT;
+    response.status.message =
+        absl::StrCat("Requested trajectory with ID ", request.trajectory_id,
+                     " doesn't exist.");
+    return;
+  } else if (trajectory_states.at(request.trajectory_id) ==
+             cartographer::mapping::PoseGraphInterface::TrajectoryState::
+                 DELETED) {
+    response.status.code =
+        cartographer_ros_msgs::StatusCode::RESOURCE_EXHAUSTED;
+    response.status.message =
+        absl::StrCat("Requested trajectory with ID ", request.trajectory_id,
```

Comment:
self-review: not much value of having an additional code path, can be simplified by combining with the other check to "Requested trajectory with ID x doesn't exist or has been deleted."